pkg/utils/utils.go

```diff
@@ -238,8 +240,8 @@ func HasDpdkDriver(pciAddr string) (bool, error) {
 
 // SaveNetConf takes in container ID, data dir and Pod interface name as string and a json encoded struct Conf
 // and save this Conf in data dir
-func SaveNetConf(cid, dataDir, podIfName string, conf interface{}) error {
-	netConfBytes, err := json.Marshal(conf)
+func SaveNetConf(cid, dataDir, podIfName string, netConf sriovtypes.NetConf) error {
```

Comment:
so we use the concrete type instead of ptr to not call [1] ?

https://github.com/containernetworking/cni/blob/c04330ee9e0bb932e1e3eda757813e2f212e746b/pkg/types/types.go#L86

Reply:
feels like we should implement our own MarshalJSON ?

Reply:
I need to check if this will introduce an issue with DNS I will create a unit-test to cover this one

Reply:
Hi please take a look I think a better solution is to just do the same DNS check as part of our save config